containers/src/Data/Graph.hs

```diff
@@ -220,11 +220,18 @@ instance F.Foldable SCC where
   foldr c n (AcyclicSCC v) = c v n
   foldr c n (NECyclicSCC vs) = foldr c n vs
 
+  null _ = False
```

Comment:
I imagine GHC can figure this out from the default, but I agree it's nice to make it clear.

Reply:
Apparently it doesn't. However, I realized that we're making `null` lazier here. The default definition using `foldr` would need to inspect the `SCC`. Is this the right thing to do? I'm not sure, though many `Foldable` instances in base do this.

Reply:
Ugh. I hate those kinds of decisions. I think the change is probably fine, but must be logged. `null` isn't normally used to force computations.

Reply:
After some thought, I feel it's best to let this lie. It is very unlikely someone is using `null` on an `SCC`, so this is unlikely to help someone with performance. But in case they are, this is an unnecessary breaking change to deal with.  
Let me know if you feel otherwise.